walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,102 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  /**
+   * Computes a list of all time ranges where all attendees of a meeting are busy
+   * @param  events    Collection of Event objects formed before the creation of this meeting
+   * @param  attendees Collection of attendee string names
+   * @return List of time ranges where attendees are engaged in other events.
+   */
+  private List<TimeRange> getBusyTimeRanges(Collection<Event> events, Collection<String> attendees) {
+    List<TimeRange> busyTimeRanges = new ArrayList<TimeRange>();
+    for (Event event : events) {
+      Set<String> currEventAttendees = event.getAttendees();
+      for (String attendee : attendees) {
+        if (currEventAttendees.contains(attendee)) {
+          busyTimeRanges.add(event.getWhen());
+          break;
+        }
+      }
+    }
+    return busyTimeRanges;
+  }
+
+  /**
+   * Removes nested and overlapping time ranges in a certain sorted List 
+   * of TimeRange objects
+   * @param  sortedtimeRanges  List of TimeRange objects that must be ordered by starting time
+   */
+  private void discretizeSortedTimeRanges(List<TimeRange> sortedTimeRanges) {
+    ListIterator<TimeRange> iter = sortedTimeRanges.listIterator();
+    TimeRange prevTimeRange = null;
+
+    while (iter.hasNext()) {
+      TimeRange curr = iter.next();
+      if (prevTimeRange != null) {
+        if (prevTimeRange.contains(curr)) {
+          iter.remove();
+        } else if (prevTimeRange.overlaps(curr)) {
+          int end = Math.max(prevTimeRange.end(), curr.end());
+          iter.set(TimeRange.fromStartEnd(prevTimeRange.start(), end, false));
+        }
+      }
+      prevTimeRange = curr;
+    }
+  }
+
+  /**
+  * Adds a time range with certain start, end, duration, and inclusitivity attributes to a Collection of 
+  * TimeRange objects if they fit the meeting requirements and are valid within a day.
+  */
+  private void addTimeRangeIfValid(Collection<TimeRange> availableTimeRanges, int start, int end, long duration,
+      boolean inclusive) {
```

Comment:
It seems like having too many parameters? Can you create the TimeRange outside the method? It will reduce the # of parameters without changing the purpose of the method.